apps/re_integrations/lib/orulo/payload_processor/typologies.ex

```
@@ -0,0 +1,47 @@
+defmodule ReIntegrations.Orulo.PayloadProcessor.Typologies do
+  @moduledoc """
+  Module to process typologies payloads into units.
+  """
+  alias Ecto.Multi
+
+  alias Re.{
+    Developments,
+    Units
+  }
+
+  alias ReIntegrations.{
+    Orulo.TypologyPayload,
+    Orulo.TypologyMapper,
+    Repo
+  }
+
+  def process_typologies(multi, typology_uuid) do
+    typology_payload = Repo.get(TypologyPayload, typology_uuid)
+    {:ok, development} = Developments.get_by_orulo_id(typology_payload.building_id)
+
+    %{payload: %{"typologies" => typologies}} = typology_payload
+
+    multi
+    |> insert_units_from_typologies(typologies, development)
+    |> Repo.transaction()
+  end
+
+  defp insert_units_from_typologies(multi, typologies, development) do
+    Multi.run(multi, :insert_units, fn _repo, _changes ->
```

What do you expect here if one of the insertion fails?
Since you're not checking for insertion errors, you might have some fail silently.
If all should fail, should do multiple `Multi.insert`s.
If just one should fail, you should do multiple jobs.

---

I have no sure about what we could consider a failure but will think about it and offer some alternatives here.

---

Invalid payload and invalid data is pretty straight-forward.
Also for "valid" data. For example, you could have a unit with price less than `250_000` that would break insertion.

---

I'm login these cases now, in the same way, we're doing with images. Process all, info split them into successful and fail cases, log failed ones and considering the process as a success. 

When we find some failures here (probably 'cause validations), we should stop to debate if what we consider an invalid should be flexibility by our model or if it should keep the same and really ignore these cases...